### PR TITLE
ci: bump macos version to 13

### PR DESCRIPTION
**Pull request by ComradeProgrammer**
ci: bump macos version to 13

related

### CODE REVIEW

**Comment by ComradeProgrammer** [comment]
@spowelljr  @medyagh could you please type an ok-to-test?
Now I changed the value of `runs-on` to `[macos-13,x64]` according to https://docs.github.com/en/actions/using-jobs/choosing-the-runner-for-a-job 

I think it may works

**Comment by spowelljr** [comment]
The additional tags are for self-hosted runners where you specify the tags on the runner itself, I don't think this is going to work.

**Comment by ComradeProgrammer** [comment]
> The additional tags are for self-hosted runners where you specify the tags on the runner itself, I don't think this is going to work.

I see. I shall try something else later

**Comment by ComradeProgrammer** [comment]
I think this time it might work.
Basically, what I did is that adding a new workflow into the github action. This workflow use PR workflow as its dependency, so that it will be triggered once this workflow is finished.

This workflow will try to use gh commandline to check the log of its depedent PR workflow to see if there are warnings about incorrect architecture(`gh run view ${{ github.event.workflow_run.id }} --log | grep ':type=>:arm'`) .

 If such error is found in the log, then it will try to fetch the jobid of `PR / functional_virtualbox_macos ` (` JOBID=`gh run view ${{ github.event.workflow_run.id }} --json jobs | jq '.jobs[] | select ( .name == "functional_virtualbox_macos") | .databaseId'``)and rerun it(`gh run rerun --job $JOBID`)

I created a test repo and used a simplified version of it, and I think it works. However, it seems that this change cannot take effect before it is merged.

File: .github/workflows/virtual_box_rerun.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This is a basic workflow to help you get started with Actions
```
Comment (medyagh) [inline .github/workflows/virtual_box_rerun.yml:1]: add better comments and explain what is this and why we need it

**Comment by medyagh** [comment]
> mmandline to check the log of its depedent PR workflow to see if there are warnings about incorrect architecture(`gh run view ${{ github.event.workflow_run.id }} --log | grep ':type=>:arm'`) .
> 
> If such error is found in the log, then it will try to fetch the jobid of `PR / functional_vir`

cool, lets see if you can make the Test Pass too ?

File: .github/workflows/virtual_box_rerun.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  check-and-rerun:
+    runs-on: ubuntu-latest
+    permissions: write-all
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: CheckArm64AndRerun
+        run: | 
+          set +e
+          gh run view ${{ github.event.workflow_run.id }} --log | grep ':type=>:arm'
```
Comment (medyagh) [inline .github/workflows/virtual_box_rerun.yml:25]: in the Original Job, we should have a Check that checks "step" for Architecture
and in the step it should fail Nice-er

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -527,8 +527,17 @@ jobs:
...
     steps:
+      - name: Check architecture
+        shell: bash
+        run: |
+          if [ $(uname -m) != "x86_64" ] 
+          then 
+              echo "$(uname -m) detected. This test must be run on x86_64"
+              exit -1
+          fi
+          exit 0
```
Comment (medyagh) [inline .github/workflows/pr.yml:540]: if this "Exit 0" will it still be triggered by the Re-Run job

Comment (medyagh) [reply .github/workflows/pr.yml:540]: I also suggest that you create an issue in in minikube and also in Github Action itself, so they can fix this for everyone, they should not give users Arm Machines

**Review by medyagh — CHANGES_REQUESTED** [review]

**Comment by medyagh** [comment]
@ComradeProgrammer can we make sure we have create an issue in Github Action to make sure they know this issue exists?

**Comment by spowelljr** [comment]
How often was this occurring? I just tried to replicate this, I ran 50 jobs using a `macos-13` runner and all of them gave me an x86 machine. Was this possibly resolved?

**Comment by medyagh** [comment]
I think I would like to do this PR without the Re-Run logic that way at least we will have a nice message that it failed

**Comment by spowelljr** [comment]
VirtualBox doesn't support macOS 13+ so I think we can close this PR.

**Pull request closed without merging**